python/graphstorm/trainer/lp_trainer.py

```diff
@@ -131,9 +131,13 @@ def fit(self, train_loader, num_epochs,
                 pos_graph = pos_graph.to(device)
                 neg_graph = neg_graph.to(device)
                 blocks = [blk.to(device) for blk in blocks]
-                input_feats = data.get_node_feats(input_nodes, device)
                 for _, nodes in input_nodes.items():
                     num_input_nodes += nodes.shape[0]
+                input_feats = data.get_node_feats(input_nodes, device)
+
+                # Preparing edge feature for pos_graph and neg_graph if any.
+                model.module.prepare_pos_graph(pos_graph, data, device)
```

Comment:
it's a bit weird to have a model to do this. can we get the edge features in the trainer and pass them to the model?

Reply:
In prepare_pos_graph, we are copying edge weights into pos_graph to compute edge loss. This is different from passing edge features for message passing. Pos_graph and neg_graph are not used in GNN message passing. 
My understanding of "get the edge features in the trainer and pass them to the model" is that it is how we pass edge features for message-passing.

Reply:
not necessarily. Trainers are responsible to prepare mini-batch data for model computation. Here it asks models to prepare the mini-batch data.

Reply:
the other question is how it is different from preparing edge features for message passing and for decoders? we should implement them in the same way.